llama-stack-client-kotlin-client-local/src/main/kotlin/com/llama/llamastack/client/local/util/PromptFormatLocal.kt

```diff
@@ -0,0 +1,98 @@
+package com.llama.llamastack.client.local.util
+
+import com.llama.llamastack.models.CompletionMessage
+import com.llama.llamastack.models.InferenceChatCompletionParams
+import com.llama.llamastack.models.SystemMessage
+import com.llama.llamastack.models.UserMessage
+
+object PromptFormatLocal {
+    private const val SYSTEM_PLACEHOLDER: String = "{{ system_prompt }}"
+    private const val USER_PLACEHOLDER: String = "{{ user_prompt }}"
+    private const val ASSISTANT_PLACEHOLDER: String = "{{ assistant_response }}"
+
+    private fun getSystemPromptTemplate(modelName: String?): String {
+        return when (modelName) {
+            "LLAMA_3",
+            "LLAMA_3_1",
+            "LLAMA_3_2" ->
+                "<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n$SYSTEM_PLACEHOLDER<|eot_id|>"
+            else -> SYSTEM_PLACEHOLDER
+        }
+    }
+
+    private fun getUserPromptTemplate(modelName: String?): String {
+        return when (modelName) {
+            "LLAMA_3",
+            "LLAMA_3_1",
+            "LLAMA_3_2",
+            "LLAMA_GUARD_3" ->
+                "<|start_header_id|>user<|end_header_id|>$USER_PLACEHOLDER<|eot_id|><|start_header_id|>assistant<|end_header_id|>"
+            else -> USER_PLACEHOLDER
+        }
+    }
+
+    private fun getAssistantPromptTemplate(modelName: String?): String {
+        return when (modelName) {
+            "LLAMA_3",
+            "LLAMA_3_1",
+            "LLAMA_3_2" -> "\n$ASSISTANT_PLACEHOLDER<|eot_id|>"
+            else -> ASSISTANT_PLACEHOLDER
+        }
+    }
+
+    fun getStopToken(modelName: String?): String {
+        return when (modelName) {
+            "LLAMA_3",
+            "LLAMA_3_1",
+            "LLAMA_3_2",
+            "LLAMA_GUARD_3" -> "<|eot_id|>"
```

Comment:
what about `<|eom_id|>` which is a stop-token the models also emit

Reply:
<|eot_id|> is the stop token we have used for the Executorch demo app as well (https://github.com/pytorch/executorch/blob/main/examples/demo-apps/android/LlamaDemo/app/src/main/java/com/example/executorchllamademo/PromptFormat.java#L63).

<|eom_id|> is used for when `Environment: ipython` instruction is used in the system prompt which we don't support, or if the model calls for a built-in tool (we're support custom calendar tool calling in our case). So this token doesn't apply to us in this scenario.

Reply:
Sorry I don't understand this. It is absolutely a token the model emits and is fundamental to the operation of the model when it does built-in tool calling. This is core infrastructure we are setting up here and llamastack *must absolutely* support it.

Reply:
thanks @ashwinb for the feedback! I've added in `<|eom_id|>` stopToken and modified the function to be `getStopTokens()`.